### PR TITLE
Log: Implementation File.Edit

### PR DESCRIPTION
**Pull request by WingToh**
Log: File.Edit has been implemented. Log is fired for every save and upon handling diffs, it is determined which of these are Edits. CodeStateSection is included, EditType is not as the fields are way too detailed for our application of the logs.

### CODE REVIEW

File: PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs [hunk 1]
```diff
@@ -10,13 +10,15 @@
 using PatternPal.Protos;
 using System.Threading;
 using System.Collections.Generic;
+using Microsoft.VisualStudio.Shell.Interop;
```
Comment (ovda96) [inline PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs:13]: Is unused, please remove

**Comment by ovda96** [comment]
Paused review until `develop` is merged onto this branch.

File: PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs [hunk 2]
```diff
@@ -256,13 +259,27 @@ private static void OnDebugProgram(dbgEventReason reason)
...
 
         /// <summary>
         /// The event handler for when recognizing software patterns in the extension.
         /// </summary>
-        public static void OnPatternRecognized(RecognizeRequest recognizeRequest, IList<RecognizeResult> recognizeResults)
+        public static void OnPatternRecognized(RecognizeRequest recognizeRequest,
+            IList<RecognizeResult> recognizeResults)
         {
-            if (_package == null || !_package.DoLogData) return; 
+            if (_package == null || !_package.DoLogData) return;
```
Comment (ovda96) [inline PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs:282]: Isn't this caught in different places for the other events?

**Review by ovda96 — CHANGES_REQUESTED** [review]

File: PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs (hunk 2 above)
Comment (WingToh) [reply PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs:282]: No, this method is called from the detector view which is initialized earlier than the package. This was also already on develop.

File: PatternPal/PatternPal.sln [hunk 3]
```diff
@@ -8,6 +8,9 @@ EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "10 - Back-end", "10 - Back-end", "{25C2E758-AC65-4A3A-A11A-2D3510C01BB8}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "20 - Front-end", "20 - Front-end", "{D8082781-8981-4B17-9E4A-5C25EF7E3892}"
+	ProjectSection(SolutionItems) = preProject
```
Comment (ovda96) [inline PatternPal/PatternPal.sln:11]: Again; this should be removed. This is wat causes `IResult.cs` to show op separately in the solution explorer.

**Review by ovda96 — CHANGES_REQUESTED** [review]
There is still an issue with the solution file but I'll commit a fix myself :).

**Pull request merged**